nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -232,6 +232,7 @@ class RegulatedPurePursuitController : public nav2_core::Controller
   std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmap_ros_;
   nav2_costmap_2d::Costmap2D * costmap_;
   rclcpp::Logger logger_ {rclcpp::get_logger("RegulatedPurePursuitController")};
+  rclcpp::Clock::SharedPtr clock_;
```

Comment:
Below `  logger_ = node->get_logger();` in L53 place the get_clock so that we get the clock from the node. Right now I don't think you initialize this anywhere...